[Menu] Add a dividerStyle property

gabrielmdeal opened the pull request:
- [x] PR has tests / docs demo, and is linted.
  * Linted -- **but no tests or docs**.  I can add this if you think it is justified.
- [x] Commit and PR titles begin with [ComponentName], and are in imperative form: "[Component] Fix leaky abstraction".
- [x] Description explains the issue / use-case resolved, and auto-closes the related issue(s) (http://tr.im/vFqem).


Add a dividerStyle property that allows users to customize the style on any Divider children of the Menu.

oliviertassinari commented:
I'm wondering about this new property. Wouldn't it be better if users were able to provide a `style` like this:
```jsx
  <Menu>
    <MenuItem primaryText="Settings" />
    <MenuItem primaryText="Help & feedback" />
    <Divider style={style} />
    <MenuItem primaryText="Sign out" />
  </Menu>
```

> I can add this if you think it is justified.

The `Menu` component don't have unit test yet. So I definitely think that it would be great to start somewhere. I understand if you don't know enzyme and have any clue where to start.

gabrielmdeal commented:
@oliviertassinari That does seem like the best way to do it!  It is how I initially attempted to style my `Divider`.  But currently the styles set on a `Divider` (as you describe) are clobbered by `Menu` when it clones the `Divider`.  

If I change this behavior, how do you think it should behave if the `Divider` already has a style set?  Options:
* Merge the styles from `Menu` into the `Divider` element's existing styles.  Like this: `mergedDividerStyles = Object.assign({}, styles.divider, child.props.styles)`
* Do not modify the `Divider` element's styles if they are set.  Only use the `Menu` element's hard-coded `style.divider` when `Divider.props.styles` is not set.
* Something else?

Note that the two above options have the possibility of changing Material UI's behavior for existing code that sets a style on a `Divider`.  Adding a new `dividerStyle` property is 100% backwards compatible.

oliviertassinari commented:
> But currently the styles set on a Divider (as you describe) are clobbered by Menu when it clones the Divider.

True, the first option is basically what we are doing everywhere else in that case. Let's go for it 👍 .

gabrielmdeal commented:
I updated my change to use `child.props.style`.  (I squashed my old commit in the process.)  Look good?

I'll try to write a couple tests tomorrow, but it may not happen until Tuesday night.

gabrielmdeal commented:
It's ready to go as far as I know!

oliviertassinari commented:
@gabrielmdeal Thanks.
